tests/bin/difference.py

```diff
@@ -225,41 +202,140 @@ def update_command(command, new_viewer, new_reportdir):
 def update_job(job, new_viewer, new_report_root):
 
     job = copy.copy(job) # a shallow copy of job
-    command = job["--command"]
-    outputs = job["--outputs"]
-    proof_name = job["--pipeline-name"]
+    command = job["command"]
+    outputs = job["outputs"]
+    proof_name = job["pipeline_name"]
     assert len(outputs) == 1
 
     new_reportdir = form_reportdir(new_report_root, proof_name)
-    job["--command"] = update_command(command, new_viewer, new_reportdir)
-    job["--outputs"] = [new_reportdir]
+    job["command"] = update_command(command, new_viewer, new_reportdir)
+    job["outputs"] = [new_reportdir]
     return job
 
 def update_jobs(jobs, new_viewer, new_report_root):
     return [update_job(job, new_viewer, new_report_root) for job in jobs]
 
 ################################################################
 
-def run_viewer(proof_root, viewer1, root1, viewer2, root2, litani):
+def parsed_command(cmd):
+    cmd = re.sub(r'\s+', ' ', cmd.strip())
+
+    command, *options = cmd.split(' ')
+    keys = options[0::2]
+    vals = options[1::2]
+    pairs = zip(keys, vals)
+
+    return command, dict(pairs)
+
+def make_coverage(cmd, opts, old):
+    cov = opts['--coverage']
+    src = opts['--srcdir']
+    if old:
+        cmd = Path(cmd).with_name('make-coverage')
+        return f"{cmd} {cov} --srcdir {src}"
```

Comment:
This is within `tests/` so not a big deal for end users, but this function makes me nervous. It's very fragile -- expects exactly one argument (`value`) for each cmdline flag (`key`) and doesn't allow positional arguments. Would it be possible to reuse our parser here?

Also, it looks like `parsed_command` is being called to extract `old_opts` in `make_kind`, but many `make_*` functions generate positional arguments without a "key" that this function expects. For instance, the following `old` case in `make_coverage` generates a positional argument `{cov}` without a "key":
https://github.com/model-checking/cbmc-viewer/blob/90d3dbd29b02a410f51ef0f7d29536f7eb8c96fb/tests/bin/difference.py#L233-L235

Is this correct?

Reply:
Sorry, just noticed that I posted this comment on the wrong line. I am talking about the function above, `parsed_command` in my comment.